doc/book/helpers/server-url-helper.md

```diff
@@ -47,7 +47,8 @@ As such, you will need to:
 
 - Register the `ServerUrlHelper` as a service in your container.
 - Register the `ServerUrlMiddleware` as a service in your container.
-- Register the `ServerUrlMiddleware` as pre_routing pipeline middleware.
+- Register the `ServerUrlMiddleware` as pipeline middleware, immediately
+  following the routing middleware.
```

Comment:
**before**, so that you can take advantage of this one even if some error occurs in routing